cluster/deposit.go

```diff
@@ -45,3 +52,42 @@ func depositDataFromJSON(d depositDataJSON) DepositData {
 		Signature:             d.Signature,
 	}
 }
+
+// VerifyDepositAmounts verifies various conditions about partial deposits rules.
+func VerifyDepositAmounts(amounts []eth2p0.Gwei) error {
+	if len(amounts) == 0 {
+		return nil
+	}
+
+	const minSingleDepositAmountGwei = 1000000000
+	const maxTotalDepositAmountGwei = 32000000000
+
+	var sum eth2p0.Gwei
+	for _, amount := range amounts {
+		if amount < minSingleDepositAmountGwei {
+			return errors.New("each partial deposit amount must be greater than 1ETH", z.U64("amount", uint64(amount)))
+		}
+		sum += amount
+	}
+
+	if sum > eth2p0.Gwei(maxTotalDepositAmountGwei) {
+		return errors.New("sum of partial deposit amounts must sum up to 32ETH", z.U64("sum", uint64(sum)))
+	}
+
+	return nil
+}
+
+// DepositAmountsFromIntSlice converts amounts from []int to []eth2p0.Gwei.
+// For verification, please see VerifyDepositAmounts().
+func DepositAmountsFromIntSlice(amounts []int) []eth2p0.Gwei {
```

Comment:
This looks like a rather long name, rather suggest
```suggestion
func IntsToGweis(amounts []int) []eth2p0.Gwei {
```